cirq-core/cirq/qis/clifford_tableau.py

```diff
@@ -176,6 +179,49 @@ def _str_full_(self) -> str:
 
         return string
 
+    def then(self, second: 'CliffordTableau') -> 'CliffordTableau':
```

Comment:
Shouldn't we do some kind of dimensionality checking? I haven't see the cases handled/tested when the sizes of the tableaus are different.

Reply:
Thanks for catching that. (If there is no check in the beginning, it will throw an error at the matrix multiplication step. But it may be confusing.) Add the corresponding test as well.